tests/console/verify_systemd_timesync.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that systemd timer is used for time synchronization.
+#          - Check if configuration file exists and contains expected values.
+#          - Check the message logs for "One time synchronization" occurance.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+
+    record_info("Check configuration", "Check if file /etc/systemd/system/yast-timesync.timer exists and has expected configuration.");
+    assert_script_run('ls /etc/systemd/system/yast-timesync.timer');
+    my $conf_file = script_output("cat /etc/systemd/system/yast-timesync.timer");
+    $conf_file =~ /OnUnitActiveSec=(?<interval>(\S+))min/;
+    my $expected_interval = $test_data->{profile}->{'ntp-client'}->{ntp_sync};
+    assert_equals($expected_interval, $+{interval}, "The interval in yast-timesync configuration file is not the expected one.");
+
+    record_info("Check sync", "Check if the use of systemd timers can be spotted in the journal logs as expected.");
+    # The first "One time synchronization" is not happening immediately after boot. In order to make sure that first sync
+    # has already happened before checking the logs, the following loop will check the logs repeatedly until logs are found
+    # or uptime minutes are more that OnBootSec+1 ( +1 due to timers inaccuracy ).
+    $conf_file =~ /OnBootSec=(?<onbootsec>(\S+))min/;
+    my $uptime;
+    my $not_synced = 1;
+    do {
+        $not_synced = script_run("journalctl -u yast-timesync | grep \"Started One time sync configured by YaST\"");
```

Comment:
here in some way it seems you could use the `--line-buffered`. to understand how this could be helpful consider that without it you get the result from your command once the buffered if fulfilled. what if you want to follow up the output of journalctl and get the out put as fast as it get in the logs? you have to disable the buffering. And this is what the   `--line-buffered` does. 
So something that you might try is to add `journalctl -f` and `grep --line-buffered`, redirect to a file and check this while  uptime is less than $+{onbootsec} as you are doing already. not sure if this is really helpful and i dont want to complicate it. But i hope you can take some ideas.

Reply:
I will try it.

Reply:
if the event is written in the logs after your write the file and before you check the file we could miss the event, right?

Reply:
I could do something similar as the yast2_lan_gui test. There will be the process running in the background, which will be greping the "Start One time sync" logs, every time a sync occurs and will be writing into a file, so instead of checking directly the logs, I can be checking the file. The benefit would be that I will have a smaller command in the loop. On the downside this will slightly rise the complexity. Something like this:
http://pastebin.sup.scz.suse.com/view/3644adfa
It would be nice if assert_script_run with given $timeout would stop if asserted before the timeout, but unfortunately, doesn't work this way.

Reply:
it is more complex and you have to kill the process afterward. There is in the api `validate_script_output` that can adjust timeout, in case you want to play with it.
Other thing is if the uptime is enough reliable, we could just introduce a `count` variable to iterate  and a maximum variable `(OnBootSec + delta) x 60/10` due to we check every 10 seconds and delta for example 1 second and put some comment. But we are overthinking here :) You can do that change if thinks is easier or leave as it is.

Reply:
I will leave it as it is . Seems to me that the timers inaccuracy wouldn't allow calculations to help.